frontend/catalyst/debug/compiler_functions.py

```diff
@@ -160,3 +162,59 @@ def compile_from_mlir(ir, compiler=None, compile_options=None):
         result_types = [mlir.ir.RankedTensorType.parse(rt) for rt in func_data[1].split(",")]
 
     return CompiledFunction(shared_object, qfunc_name, result_types, None, compiler.options)
+
+
+@debug_logger
+def compile_cmain(fn, *args):
```

Comment:
Maybe slightly better name :)
```suggestion
def compile_executable(fn, *args):
```